### PR TITLE
docs: add example with ws route

### PR DESCRIPTION
**Pull request by zvlb**
Commit Message: A new example using the WebSocket protocol on a specific Route has been added to the documentation and example of working with WebSocket.
Risk Level: Low
Testing: A new example for automated testing has been added to the verify.sh script.
Docs Changes: example.rst has been updated
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by zvlb** [comment]
How it looks on site - https://storage.googleapis.com/envoy-pr/6777f12/docs/start/sandboxes/websocket.html

And I don't know why step `Envoy/Publish and verify` fieled in actions

File: examples/websocket/example.rst [hunk 1]
```diff
@@ -49,7 +49,7 @@ Step 2: Build and start the sandbox
 
 This starts three proxies listening on ``localhost`` ports ``10000-30000``.
 
-It also starts two upstream services, one ``ws`` and one ``wss``.
+It also starts four upstream services, ``ws``, ``ws-route``, ``wss`` and ``wss-passthrough``.
```
Comment (phlax) [inline examples/websocket/example.rst:52]: the original is correct here - this is about _upstream_ services - ie backend websockets that envoy connects to.

Comment (phlax) [reply examples/websocket/example.rst:52]: The bit that needs to change is above

> This starts three proxies listening on ``localhost`` ports ``10000-30000``.

File: examples/websocket/example.rst [hunk 2]
```diff
@@ -99,7 +100,42 @@ You can start an interactive session with the socket as follows:
...
+In order for Envoy to terminate the WebSocket connection, the
+:ref:`upgrade_configs <envoy_v3_api_msg_config.route.v3.RouteAction.UpgradeConfig>`
+in :ref:`RouteAction <envoy_v3_api_msg_config.route.v3.RouteAction>`
+must be set, as can be seen in the provided :download:`ws -> ws configuration <_include/websocket/envoy-ws-route.yaml>`:
+
+.. literalinclude:: _include/websocket/envoy-ws-route.yaml
+   :language: yaml
+   :lines: 1-35
+   :linenos:
+   :emphasize-lines: 24-25
```
Comment (phlax) [inline examples/websocket/example.rst:118]:
```suggestion
   :language: yaml
   :lines: 19-25
   :linenos:
   :emphasize-lines: 6-7 
   :caption: :download:`envoy-ws-route.yaml <_include/websocket/envoy-ws-route.yaml>`
   
```


rather show a short snippet with highlight and let the user download the full file more easily

**Review by phlax — COMMENTED** [review]
@zvlb thanks for this - really, really cool addition

**Comment by phlax** [comment]
> And I don't know why step Envoy/Publish and verify fieled in actions

i just landed a (hopeful) fix/workaround

File: examples/websocket/example.rst (hunk 2 above)
Comment (zvlb) [reply examples/websocket/example.rst]: fixed

File: examples/websocket/example.rst (hunk 1 above)
Comment (zvlb) [reply examples/websocket/example.rst]: Hm. What need to change in this line? New test started on port 15000 (it's between 10000-30000).
May be need to start tests on 10000-20000-30000-40000 ports? (In my PR: 10000-15000-20000-30000)

Comment (phlax) [reply examples/websocket/example.rst]: > This starts four proxies listening on localhost ports 10000, 15000, 2000, 30000.

**Comment by phlax** [comment]
apologies - my fix broke formatting - hopefully should work now

File: examples/websocket/docker-compose.yaml [hunk 3]
```diff
@@ -19,14 +19,23 @@ services:
...
     - "${PORT_PROXY0:-10000}:10000"
 
+  proxy-ws-route:
+    build:
+      context: .
+      dockerfile: ../shared/envoy/Dockerfile
+      args:
+        ENVOY_CONFIG: ./envoy-ws-route.yaml
+    ports:
+    - "${PORT_PROXY1:-15000}:10000"
```
Comment (phlax) [inline examples/websocket/docker-compose.yaml:29]:
```suggestion
    - "${PORT_PROXY1:-15000}:15000"
```

Comment (zvlb) [reply examples/websocket/docker-compose.yaml:29]: I'm not sure... What need to do?

Comment (phlax) [reply examples/websocket/docker-compose.yaml:29]: nothing - my mistake - all good here

**Review by phlax — APPROVED** [review]
lgtm, thanks again @zvlb

**Pull request merged**